### PR TITLE
Docker test completion

### PR DESCRIPTION
**Pull request by dakshhhhh16**
## Summary

This PR completes the Phase 1 development environment and test foundation work.

It makes the project reproducible from a fresh checkout, adds Docker-based startup for the full stack, improves seed/reset workflows, adds deterministic local sample data, and introduces core backend tests for healthcare desert scoring and affordability burden logic.

It also fixes Alaska community coordinate seeding so known communities are plotted from corrected coordinates instead of stale transport-profile values.

### Dockerized Development Workflow

- Added container support for the Flask backend.
- Added a multi-stage frontend Docker build:
  - Node build stage for `npm install` and `npm run build`
  - nginx stage for serving optimized React assets
- Added Docker Compose orchestration for backend and frontend services.
- Added environment wiring through `.env` / `.env.example`.
- Added `.dockerignore` coverage for generated files, caches, virtual environments, uploads, and local database artifacts.

### Seed Data And Local Reproducibility

- Added offline sample Census income data for affordability calculations.
- Added offline sample Ookla performance data for broadband/performance routes.
- Added corrected transport-profile coordinate usage for Alaska communities.
- Prevented repeated seed runs from duplicating CAT gating rules.

### Healthcare Desert Scoring

- Added healthcare desert scoring support for:
  - distance to nearest facility
  - facility density
  - specialist availability
  - transport accessibility
  - seasonal adjustment
  - composite score calculation
- Added focused pytest coverage for documented score formulas and edge cases.

### Affordability Burden Logic

- Added affordability tests for:
  - ISP pricing lookups
  - Starlink pricing bucket handling
  - ZCTA income matching
  - burden percentage calculation
  - ITU 2% threshold boundary behavior

## Validation

The following checks passed locally:

- `python3 -m pytest -q`
- `npm run build`
- `docker compose config`

Expected pytest result:

```text
27 passed, 1 warning
```
1. Backend pytest passing
image
2. Frontend production build passing
image

3. Docker Compose config validating successfully
image
4. Clean PR commit comparison showing only this branch’s intended commits
image

### CODE REVIEW

File: backend/services/healthcare_desert_calculator.py [hunk 1]
```diff
@@ -46,55 +46,131 @@ def calculate_distance(lat1: float, lon1: float, lat2: float, lon2: float) -> fl
...
+
+        Returns clinic, hospital, and nearest-facility distances in kilometers.
+        """
+        center = HealthcareDesertCalculator._region_center(db, region_code)
+        if not center:
+            return {"clinic": None, "hospital": None, "nearest": None}
+
+        sites = db.query(HealthcareSite).filter(
+            HealthcareSite.is_active == True
+        ).all()
```
[the remark this answers is not in the document]
Comment (dakshhhhh16) [reply backend/services/healthcare_desert_calculator.py:141]: This was needed
